lib/rules/no-prefix-styled-component.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @fileoverview Prevent the declaration of a styled component prop without the dollar sign prefix
+ * @author Max Balej
+ */
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: "problem",
+    docs: {
+      description:
+        "Prevent the declaration of a styled component prop without the dollar sign prefix",
+      category: "Wrong component usage",
+      recommended: true,
+    },
+    messages: {
+      prefixError: "Expected prefix '$' for styled-components props.",
+    },
+    fixable: null,
+    schema: [],
+  },
+
+  create: function (context) {
+    const styleIdentifier = "styled";
+
+    function unwrapExpression(node) {
```

Comment:
? (le nom ne m'a pas tout de suite parlé/indiqué que ça retournait un boolean)
```suggestion
    function isStyledExpression(node) {
```